xds/internal/xdsclient/watchers.go

```diff
@@ -27,31 +27,39 @@ import (
 // calling cancel()), there's a small window where the callback can be called
 // after the watcher is canceled. The caller needs to handle this case.
 func (c *clientImpl) WatchListener(serviceName string, cb func(xdsresource.ListenerUpdate, error)) (cancel func()) {
-	first, cancelF := c.pubsub.WatchListener(serviceName, cb)
-	if first {
-		c.controller.AddWatch(xdsresource.ListenerResource, serviceName)
+	n := xdsresource.ParseName(serviceName)
+	name := n.String()
```

Comment:
Optional: Get rid of `name` and simply inline `n.String()` in the call to `a.watchXxxx()`?

Reply:
Done